packages/react-core/src/components/Page/Page.tsx

```diff
@@ -110,15 +135,20 @@ export class Page extends React.Component<PageProps, PageState> {
     this.state = {
       desktopIsNavOpen: managedSidebarOpen,
       mobileIsNavOpen: false,
-      mobileView: false
+      mobileView: false,
+      width: 0
     };
   }
 
   componentDidMount() {
-    const { isManagedSidebar, onPageResize } = this.props;
-    if (isManagedSidebar || onPageResize) {
+    const { isManagedSidebar, onPageResize, useResizeObserver } = this.props;
+    if (isManagedSidebar || onPageResize || useResizeObserver) {
       if (canUseDOM) {
-        window.addEventListener('resize', this.handleResize);
+        if (useResizeObserver) {
+          this.observer = getResizeObserver(this.pageRef.current, this.handleResize);
+        } else {
+          window.addEventListener('resize', this.handleResize);
```

Comment:
Curious why we offer `useResizeObserver` as a choice? 

Note that `getResizeObserver` will use a window event listener if `containerRefElement` is not available.

Reply:
hmm, I suppose we can just enable the observer by default without providing a prop. And will update the code to remove the extra listener, thanks!